CHANGELOG..md

```diff
@@ -0,0 +1,108 @@
+<!--
+Guiding Principles:
+
+Changelogs are for humans, not machines.
+There should be an entry for every single version.
+The same types of changes should be grouped.
+Versions and sections should be linkable.
+The latest version comes first.
+The release date of each version is displayed.
+Mention whether you follow Semantic Versioning.
+
+Usage:
+
+Change log entries are to be added to the Unreleased section under the
```

Comment:
For Unreleased section, statement should be a separate line.  Unreleased section should include upcoming PR details.  

And other released entries should follow the way it is written (<tag>) \#<issue-number>.

Reply:
added # to entries, Unreleased should contain all those entries yes, but since none are currently on main, I can't add them here yet. The respective pr's should rebase or cherry pick this PR's commit and add their own note to unreleased